examples/user_defined_iface.yml

```diff
@@ -0,0 +1,43 @@
+---
+user-defined:
```

Comment:
In NM we call these "generic", what's the reason to use a different term in Nmstate?

Reply:
If the items have `type: interface`, then `generic-types` could be an alternative name for `user-defined`:

```
generic-types:
    - type: interface
      name: vxcan
      handler-scripts: ...
```

Reply:
The `generic-types` cannot explains itself without document, but `user-defined` can.

Reply:
Making it self-explanatory is a good idea, I don't see this with `user-defined` as well, though. In general the whole desired state is "user-defined", since it is what the user defines as the desired state.

How about using something like `plugins:` or `addons:` instead?